File: 628-maximum-product-of-three-numbers/maximum-product-of-three-numbers.cpp
class Solution {
public:
    int maximumProduct(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int n = nums.size();

        // Case 1: three largest numbers
        long long product1 = (long long)nums[n - 1] * nums[n - 2] * nums[n - 3];

        // Case 2: two smallest (most negative) + one largest
        long long product2 = (long long)nums[0] * nums[1] * nums[n - 1];

        return (int)max(product1, product2);
    }
};

File: 628-maximum-product-of-three-numbers/maximum-product-of-three-numbers.py
class Solution:
    def maximumProduct(self, nums: list[int]) -> int:
        nums.sort()
        n = len(nums)

        # Case 1: three largest numbers
        product1 = nums[n - 1] * nums[n - 2] * nums[n - 3]

        # Case 2: two smallest (most negative) + one largest
        # Two negatives multiply to a large positive
        product2 = nums[0] * nums[1] * nums[n - 1]

        return max(product1, product2)